modules/lfs/content_store.go

```diff
@@ -106,15 +122,21 @@ func (s *ContentStore) Exists(meta *models.LFSMetaObject) (bool, error) {
 }
 
 // Verify returns true if the object exists in the content store and size is correct.
-func (s *ContentStore) Verify(meta *models.LFSMetaObject) (bool, error) {
-	p := meta.RelativePath()
+func (s *ContentStore) Verify(pointer *Pointer) (bool, error) {
+	p := relativePath(pointer)
 	fi, err := s.ObjectStorage.Stat(p)
-	if os.IsNotExist(err) || (err == nil && fi.Size() != meta.Size) {
+	if os.IsNotExist(err) || (err == nil && fi.Size() != pointer.Size) {
 		return false, nil
 	} else if err != nil {
-		log.Error("Unable stat file: %s for LFS OID[%s] Error: %v", p, meta.Oid, err)
+		log.Error("Unable stat file: %s for LFS OID[%s] Error: %v", p, pointer.Oid, err)
 		return false, err
 	}
 
 	return true, nil
 }
+
+// ReadMetaObject will read a models.LFSMetaObject and return a reader
+func ReadMetaObject(pointer *Pointer) (io.ReadCloser, error) {
```

Comment:
May need to consider cancellation here and elsewhere in this file - but that probably can wait for another pr.